check/src/kindcheck.rs

```diff
@@ -143,7 +120,7 @@ impl<'a> KindCheck<'a> {
     fn builtin_kind(&self, typ: BuiltinType) -> RcKind {
         match typ {
             BuiltinType::String | BuiltinType::Byte | BuiltinType::Char | BuiltinType::Int |
-            BuiltinType::Float | BuiltinType::Unit => self.type_kind(),
+            BuiltinType::Float | BuiltinType::Unit | BuiltinType::Type => self.type_kind(),
```

Comment:
[GHC Haskell does something similar](https://ghc.haskell.org/trac/ghc/wiki/DependentHaskell/Phase1#Kindsandtypesarethesame), and they claim that (in their own case at least) it is consistent.